test/common.js

```diff
@@ -103,6 +105,35 @@ class TestTools {
 
 		return new Sim.Battle(battleOptions);
 	}
+
+	/**
+	 * Saves the log of the given battle as a bare-bones replay file in the `test\replays` directory
+	 * You can view the replay by opening the file in any browser or by dragging and dropping the
+	 * file into a PS! client window.
+	 *
+	 * @param {Sim.Battle} battle
+	 * @param {string} [fileName]
+	 */
+	saveReplay(battle, fileName) {
+		const battleLog = battle.getDebugLog();
+		if (!fileName) fileName = 'test-replay';
+		const filePath = path.resolve(__dirname, `./replays/${fileName}-${Date.now()}.html`);
+		const out = fs.createWriteStream(filePath, {flags: 'a'});
+		out.on('open', () => {
+			out.write(`<!DOCTYPE html>
+			<meta charset="utf-8" />
+			<!-- version 1 -->
+			
+			<div class="wrapper replay-wrapper" style="max-width:1180px;margin:0 auto">
+			<div class="battle"></div><div class="battle-log"></div><div class="replay-controls"></div><div class="replay-controls-2"></div>
+			<script type="text/plain" class="battle-log-data">${battleLog}</script>
+			</div>
+			<script>
+			let daily = Math.floor(Date.now()/1000/60/60/24);document.write('<script src="https://play.pokemonshowdown.com/js/replay-embed.js?version'+daily+'"></'+'script>');
+			</script>`);
```

Comment:
```suggestion
			out.write(
				`<!DOCTYPE html>\n` +
				`<script type="text/plain" class="battle-log-data">${battleLog}</script>\n` +
				`<script src="https://play.pokemonshowdown.com/js/replay-embed.js"></script>`
			);
```

I pushed some code to simplify this: https://github.com/smogon/pokemon-showdown-client/commit/d44b94e8414d02141ad072b465fe7dcce3254c5d